### PR TITLE
Add steps for upgrading external database

### PR DESCRIPTION
**Pull request by bangelic**
Steps were added for upgrading external database from RHEL 7 to RHEL 8 while upgrading Satellite versions.

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.4/Katello 4.6
* [x] Foreman 3.3/Katello 4.5
* [x] Foreman 3.2/Katello 4.4
* [x] Foreman 3.1/Katello 4.3
* For Foreman 3.0 or older, please create a separate PR.
* We do not accept PRs for Foreman 2.3 or older.

### CODE REVIEW

File: guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_parent.adoc [hunk 1]
```diff
@@ -24,3 +24,23 @@ You cannot upgrade {ProjectServer} from a RHEL base system that is not operating
...
+Create a new {RHEL} 8 based host for PostgreSQL server that follows the external database on {RHEL} 8 documentation.
+
+. Copy the backup database to the new host.
+. Restore the backup database on the new host.
+
+If you keep the same hostname on the new PostgreSQL server, you need to update the DNS preventing the need to do  updates on {Project} itself.
+If the new PostgreSQL server has a new host name or a different CNAME, you need to run the following on the {Project}:
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# satellite-installer \
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_parent.adoc:41]: Please do not use downstream terminology. We have attributes for this.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_parent.adoc [hunk 2]
```diff
@@ -24,3 +24,23 @@ You cannot upgrade {ProjectServer} from a RHEL base system that is not operating
 To run {ProjectServer} on a {RHEL} base system operating in FIPS mode, you must install {Project} on a freshly provisioned RHEL base system operating in FIPS mode.
 For more information, see {InstallingServerDocURL}Preparing_your_Environment_for_Installation_{project-context}[Preparing your environment for installation] in _{InstallingServerDocTitle}_.
 endif::[]
+
+.Upgrade External Database
+ifdef::satellite[]
+You can upgrade an external database from {RHEL} 7 to {RHEL} 8 while upgrading {Project} from {ProjectVersionPrevious} to {ProjectVersion}.
+
+Create a new {RHEL} 8 based host for PostgreSQL server that follows the external database on {RHEL} 8 documentation.
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_parent.adoc:32]: This sounds like a prerequisite to me.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_parent.adoc [hunk 3]
```diff
@@ -24,3 +24,23 @@ You cannot upgrade {ProjectServer} from a RHEL base system that is not operating
...
+
+. Copy the backup database to the new host.
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_parent.adoc:34]: Maybe add `.Procedure` before? Maybe move this altogether to a new file.

**Review by maximiliankolb — CHANGES_REQUESTED** [review]

File: guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+You can upgrade an external database from {RHEL} 7 to {RHEL} 8 while upgrading {Project} from {ProjectVersionPrevious} to {ProjectVersion}.
+
+.Prerequisites
+Create a new {RHEL} 8 based host for PostgreSQL server that follows the external database on {RHEL} 8 documentation.
+
+.Procedure
+. Copy the backup database to the new host.
+. Restore the backup database on the new host.
+
+If you keep the same hostname on the new PostgreSQL server, you need to update the DNS preventing the need to do  updates on {Project} itself.
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc:13]:
```suggestion
If you keep the same hostname on the new PostgreSQL server, you need to update the DNS preventing the need to do updates on {Project} itself.
```

I am unsure I understand this correctly. Is this a prerequisite?

File: guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+If the new PostgreSQL server has a new host name or a different CNAME, you need to run the following on the {Project}:
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc:14]:
```suggestion
If the new PostgreSQL server has a new host name or a different CNAME, you need to run the following on {Project}:
```

Maybe reword to "update the reference to the external database on {Project}" or similar? It should IMHO also be a step (aka. start with ". ").

**Review by maximiliankolb — COMMENTED** [review]
Is this procedure unsupported for Foreman/Katello?

File: guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc (hunk 4 above)
Comment (bangelic) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc:13]: Wasn't sure if this needed to be a step but it's not a prerequisite. Since it is telling the user to update the DNS as a result of keeping the host name, I am changing it to "."

**Comment by bangelic** [comment]
> Is this procedure unsupported for Foreman/Katello?

@ehelms can you advise?

File: guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc [hunk 6]
```diff
@@ -0,0 +1,20 @@
+[id="Upgrading_External_Database_{context}"]
+= Upgrading External Database
+
+You can upgrade an external database from {RHEL} 7 to {RHEL} 8 while upgrading {Project} from {ProjectVersionPrevious} to {ProjectVersion}.
+
+.Prerequisites
+Create a new {RHEL} 8 based host for PostgreSQL server that follows the external database on {RHEL} 8 documentation.
+
+.Procedure
+. Copy the backup database to the new host.
```
Comment (ehelms) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc:10]: I am thinking we should mention as part of prerequisites that they will need to take a backup as part of this process and should consider a maintenance window to do so.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+. Restore the backup database on the new host.
+. If you keep the same hostname on the new PostgreSQL server, you need to update the DNS preventing the need to do updates on {Project} itself.
+. If the new PostgreSQL server has a new host name or a different CNAME, update the reference to the external database on {Project}:
```
Comment (ehelms) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc:13]: Points 12 and 13 should come before the backup and restore part to get customers thinking about the network setup of their new host and how it affects the upgrade. Maybe even in the prerequisities?

**Comment by bangelic** [comment]
@ekohl or @Lennonka could one of you take a look at this and see if it's ready to be merged? Thank you in advance!

File: guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+.Procedure
+. If you keep the same hostname on the new PostgreSQL server, you need to update the DNS preventing the need to do updates on {Project} itself.
+. If the new PostgreSQL server has a new host name or a different CNAME, update the reference to the external database on {Project}:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# {foreman-installer} \
+  --foreman-db-host newpostgres.example.com \
+  --katello-candlepin-db-host newpostgres.example.com \
+  --foreman-proxy-content-pulpcore-postgresql-host newpostgres.example.com \
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc:22]:
```suggestion
--foreman-db-host newpostgres.example.com \
--katello-candlepin-db-host newpostgres.example.com \
--foreman-proxy-content-pulpcore-postgresql-host newpostgres.example.com
```

**Review by maximiliankolb — COMMENTED** [review]
For new content, please add `guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc` as module to `guides/common/modules`.

I am also thinking about rewording the title to "Upgrading the External Database" because it's only one or "Upgrading External Databases" if it's multiple DBs.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+= Upgrading External Database
+
+You can upgrade an external database from {EL} 7 to {EL} 8 while upgrading {Project} from {ProjectVersionPrevious} to {ProjectVersion}.
+
+.Prerequisites
+. Create a backup.
++
+* On a virtual machine, take a snapshot.
+* On a physical machine, create a backup.
+. Create a new {EL} 8 based host for PostgreSQL server that follows the external database on {EL} 8 documentation.
```
Comment (ekohl) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc:11]: Can you add a link to that section so users don't need to search for it?

File: guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc [hunk 10]
```diff
@@ -0,0 +1,25 @@
+[id="Upgrading_External_Database_{context}"]
+= Upgrading External Database
+
+You can upgrade an external database from {EL} 7 to {EL} 8 while upgrading {Project} from {ProjectVersionPrevious} to {ProjectVersion}.
+
+.Prerequisites
+. Create a backup.
++
+* On a virtual machine, take a snapshot.
```
Comment (ekohl) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc:9]: Does this make sense? You're creating a backup anyway (to transfer to the new machine) and you're not making changes to the old host. So why is a snapshot needed?

In other words, move "Create a backup" from a Prerequisite to a step in the Procedure.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# {foreman-installer} \
+  --foreman-db-host newpostgres.example.com \
+  --katello-candlepin-db-host newpostgres.example.com \
+  --foreman-proxy-content-pulpcore-postgresql-host newpostgres.example.com \
+----
+. Copy the backup database to the new host.
+. Restore the backup database on the new host.
```
Comment (ekohl) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc:25]: Would it make sense to give explicit instructions for this? Like how to use `pg_dump` and `pg_restore`.

**Review by ekohl — CHANGES_REQUESTED** [review]
Should there be a recommendation around decomissioning the old machine?

Another thing I'd suggest is how to minimize data loss.

Let's say you have these steps:
1. Create a backup
2. Import backup
3. Reconfigure Foreman

Now anything that happens between step 1 and step 3 is lost. This can be prevented by stopping all services on Foreman as step 0.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc (hunk 11 above)
Comment (bangelic) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc:25]: @ehelms your thoughts?

File: guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc (hunk 9 above)
Comment (bangelic) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc:11]: @ekohl I am not 100% sure if this is the correct link but I am adding this:
(https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/deploying_different_types_of_servers/using-databases#migrating-to-a-rhel-8-version-of-postgresql_using-postgresql)

File: guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc (hunk 11 above)
Comment (ehelms) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_external_database.adoc:25]: The stance has been that if a user under takes running an external database server they are expected to take care of doing a backup and restore and the details of those procedures is out of scope of our documentation. Our intent with including an OS upgrade guide is more to give guidance the order of operations and general steps a user should take but the details are up to them based on their database.

**Comment by bangelic** [comment]
Is this ready to be merged?

**Review by adamlazik1 — COMMENTED** [review]
LGTM from a writer perpective, but I cannot confirm the actual content of the changes. Leaving to @ekohl for review.

File: guides/common/modules/proc_upgrading-external-database.adoc [hunk 12]
```diff
@@ -0,0 +1,25 @@
+[id="Upgrading_the_External_Database_{context}"]
+= Upgrading the External Database
+
+You can upgrade an external database from {EL} 7 to {EL} 8 while upgrading {Project} from {ProjectVersionPrevious} to {ProjectVersion}.
+
+.Prerequisites
+. On a virtual machine, take a snapshot.
+. On a physical machine, create a backup.
+. Create a new {EL} 8 based host for PostgreSQL server that follows the external database on {EL} 8 documentation.
```
Comment (Lennonka) [inline guides/common/modules/proc_upgrading-external-database.adoc:9]: Use a bulleted list (`*`) for prerequisites.

File: guides/common/modules/proc_upgrading-external-database.adoc [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+For more information, see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/deploying_different_types_of_servers/using-databases#migrating-to-a-rhel-8-version-of-postgresql_using-postgresql[Migrating to a RHEL 8 version of PostgreSQL].
+
+.Procedure
+. Create a backup.
+. If you keep the same hostname on the new PostgreSQL server, you need to update the DNS preventing the need to do updates on {Project} itself.
```
Comment (Lennonka) [inline guides/common/modules/proc_upgrading-external-database.adoc:14]:
```suggestion
. If you keep the same hostname on the new PostgreSQL server, update the DNS record for the host so that ...
```
The meaning of the second sentence is unclear to me. Can you be more specific?
Is this about updating DNS so that {Project} can still see the host as registered? Or what would happen if I didn't update the DNS?

**Review by Lennonka — CHANGES_REQUESTED** [review]
A couple of details. + Adam's comment.

File: guides/common/modules/proc_upgrading-external-database.adoc [hunk 14]
```diff
@@ -0,0 +1,25 @@
+[id="Upgrading_the_External_Database_{context}"]
```
Comment (maximiliankolb) [inline guides/common/modules/proc_upgrading-external-database.adoc:1]: Please adjust the file name to `proc_upgrading-the-external-database.adoc`

File: guides/doc-Upgrading_and_Updating/master.adoc [hunk 15]
```diff
@@ -67,6 +67,11 @@ ifdef::katello,orcharhino,satellite[]
 include::topics/upgrading_content_hosts.adoc[leveloffset=+2]
 endif::[]
 
+ifdef::foreman-el,katello,satellite[]
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/master.adoc:70]:
```suggestion
ifdef::foreman-el,katello,satellite,orcharhino[]
```

File: guides/doc-Upgrading_and_Updating/master.adoc [hunk 16]
```diff
@@ -64,6 +64,11 @@ ifdef::katello,orcharhino,satellite[]
 include::topics/upgrading_content_hosts.adoc[leveloffset=+2]
 endif::[]
 
+ifdef::foreman-el,katello,satellite,orcharhino[]
+// Upgrading the External database
```
Comment (ekohl) [inline guides/doc-Upgrading_and_Updating/master.adoc:68]: Does this comment really add anything? I know we have it all over the place, but it's essentially repeating the self-explaining filename.

File: guides/common/modules/proc_upgrading-the-external-database.adoc [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+
+.Prerequisites
+* Create a new {EL} 8 based host for PostgreSQL server that follows the external database on {EL} 8 documentation.
+For more information, see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/deploying_different_types_of_servers/using-databases#migrating-to-a-rhel-8-version-of-postgresql_using-postgresql[Migrating to a RHEL 8 version of PostgreSQL].
+
+.Procedure
+. On a virtual machine, take a snapshot.
+. On a physical machine, create a backup.
+. If you keep the same hostname on the new PostgreSQL server, update the DNS record for the host to prevent the need to do updates on {Project} itself.
+. If the new PostgreSQL server has a new host name or a different CNAME, update the reference to the external database on {Project}:
```
Comment (ekohl) [inline guides/common/modules/proc_upgrading-the-external-database.adoc:14]: I think this is confusing. Let's first start at the requirement. The requirement is that the Foreman server can reach the DB server.

There are multiple ways to achieve this.

First of all, a new hostname is used and the procedure below uses the installer to change the configs accordingly.

Then the other is that the same hostname is used. To make sure it reaches the right server, you can use two techniques.

You described changing the DNS record, but this has some potential issues. Any DNS record has a TTL (Time To Live), which allows any DNS server to cache it. If you don't plan for it and you have a high TTL value (86400 seconds, or 1 day, is common) then it can take a long time for the migration to propagate. A common technique is to plan for it, lower the TTL in advance (at least $TTL) and then change it. Once it's stable it can be raised again.

However during the migration it may still resolve to the old host if you have multiple caching servers. That's why many admins actually prefer to move the IP to the new server instead. So you use some temporary IP to completely provision & prepare the server and then add the old IP while removing it from the old server (or shutting that down).

Some also combine that and do migrate to a new IP but add both IPs to the same server during the migration so the stale DNS cache isn't an issue.

In my mind those are all fairly standard operating procedures so I wonder how much of it should be written out.

File: guides/common/modules/proc_upgrading-the-external-database.adoc [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+
+You can upgrade an external database from {EL} 7 to {EL} 8 while upgrading {Project} from {ProjectVersionPrevious} to {ProjectVersion}.
+
+.Prerequisites
+* Create a new {EL} 8 based host for PostgreSQL server that follows the external database on {EL} 8 documentation.
+For more information, see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/deploying_different_types_of_servers/using-databases#migrating-to-a-rhel-8-version-of-postgresql_using-postgresql[Migrating to a RHEL 8 version of PostgreSQL].
+
+.Procedure
+. On a virtual machine, take a snapshot.
+. On a physical machine, create a backup.
```
Comment (ekohl) [inline guides/common/modules/proc_upgrading-the-external-database.adoc:12]: This should be reduced to:
```suggestion
. Create a backup.
```

This is because the procedure later states "Copy the backup database".

Additionally, many sysadmins will keep a virtual server around for a while as a backup, instead of separate backups/snapshots. In most virtualization platforms you can just shut it down anyway.

File: guides/doc-Upgrading_and_Updating/master.adoc (hunk 16 above)
Comment (bangelic) [reply guides/doc-Upgrading_and_Updating/master.adoc:68]: @ekohl I have often wondered the same thing.

File: guides/common/modules/proc_upgrading-the-external-database.adoc (hunk 17 above)
Comment (bangelic) [reply guides/common/modules/proc_upgrading-the-external-database.adoc:14]: @evgeni related to my comment above regarding this point. Your thoughts?

File: guides/common/modules/proc_upgrading-the-external-database.adoc [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+You can upgrade an external database from {EL} 7 to {EL} 8 while upgrading {Project} from {ProjectVersionPrevious} to {ProjectVersion}.
+
+.Prerequisites
+* Create a new {EL} 8 based host for PostgreSQL server that follows the external database on {EL} 8 documentation.
+For more information, see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/deploying_different_types_of_servers/using-databases#migrating-to-a-rhel-8-version-of-postgresql_using-postgresql[Migrating to a RHEL 8 version of PostgreSQL].
+
+.Procedure
+. Create a backup.
+. If you keep the same hostname on the new PostgreSQL server, update the DNS record for the host to prevent the need to do updates on {Project} itself.
+. If the new PostgreSQL server has a new host name or a different CNAME, update the reference to the external database on {Project}:
```
Comment (evgeni) [inline guides/common/modules/proc_upgrading-the-external-database.adoc:13]: If you perform any of these steps *before* restoring the DB on the new system, you will see errors, which will confuse the reader (especially if you use the Installer method and the installer will fail).

File: guides/common/modules/proc_upgrading-the-external-database.adoc [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# {foreman-installer} \
+--foreman-db-host newpostgres.example.com \
+--katello-candlepin-db-host newpostgres.example.com \
+--foreman-proxy-content-pulpcore-postgresql-host newpostgres.example.com
```
Comment (evgeni) [inline guides/common/modules/proc_upgrading-the-external-database.adoc:20]:
```suggestion
--foreman-db-host _newpostgres.example.com_ \
--katello-candlepin-db-host _newpostgres.example.com_ \
--foreman-proxy-content-pulpcore-postgresql-host _newpostgres.example.com_
```

italics to indicate things that need replacement

**Review by evgeni — CHANGES_REQUESTED** [review]

File: guides/common/modules/proc_upgrading-the-external-database.adoc (hunk 17 above)
Comment (evgeni) [reply guides/common/modules/proc_upgrading-the-external-database.adoc:13]: I think I'd slightly reword this, making it less ambiguous:
* create a backup 
* restore the backup on the new server
* If {Project} can reach the new database server via the old name, no further changes are required. Otherwise reconfigure {Project} to use the new name: <installer invocation>

All further details are left as an exercise to the reader, as we do not know how their setup looks like.

File: guides/common/modules/proc_upgrading-the-external-database.adoc [hunk 21]
```diff
@@ -0,0 +1,23 @@
+[id="Upgrading_the_External_Database_{context}"]
+= Upgrading the External Database
+
+You can upgrade an external database from {EL} 7 to {EL} 8 while upgrading {Project} from {ProjectVersionPrevious} to {ProjectVersion}.
+
+.Prerequisites
+* Create a new {EL} 8 based host for PostgreSQL server that follows the external database on {EL} 8 documentation.
```
[the remark this answers is not in the document]
Comment (bangelic) [reply guides/common/modules/proc_upgrading-the-external-database.adoc:7]: @evgeni Is there a section for using external databases or am I missing something?

File: guides/common/modules/proc_upgrading-the-external-database.adoc (hunk 19 above)
Comment (evgeni) [reply guides/common/modules/proc_upgrading-the-external-database.adoc:12]: I'd move them towards the end, see also the wording suggestion in https://github.com/theforeman/foreman-documentation/pull/1614#discussion_r1021241684

File: guides/common/modules/proc_upgrading-the-external-database.adoc (hunk 21 above)
[the remark this answers is not in the document]
Comment (evgeni) [reply guides/common/modules/proc_upgrading-the-external-database.adoc:7]: I was expecting "follows the external database documentation" to link to https://docs.theforeman.org/nightly/Installing_Server/index-katello.html#using-external-databases_foreman, yes.

File: guides/common/modules/proc_upgrading-the-external-database.adoc (hunk 17 above)
Comment (ekohl) [reply guides/common/modules/proc_upgrading-the-external-database.adoc:12]: > create a backup

Should we recommend an offline backup? So stop all services and then create a backup to prevent data loss?

Comment (evgeni) [reply guides/common/modules/proc_upgrading-the-external-database.adoc:12]: Given https://github.com/theforeman/foreman-documentation/blob/a9d7313f01d8fbe0d7a4a88db72583a24d1f7385/guides/common/assembly_using-external-databases.adoc#L10-L12
I would tend to say "nah, not our pint"

(but we probably should add a similar sentence to non-Satellite builds of the documentation)

File: guides/common/modules/proc_upgrading-the-external-database.adoc (hunk 21 above)
[the remark this answers is not in the document]
Comment (bangelic) [reply guides/common/modules/proc_upgrading-the-external-database.adoc:7]: Thanks! It's in the Installation Guide so I have to use a different link than xref.

File: guides/common/modules/proc_upgrading-the-external-database.adoc (hunk 19 above)
Comment (bangelic) [reply guides/common/modules/proc_upgrading-the-external-database.adoc:12]: There really wasn't a place to move these since there weren't many steps in the Procedure to begin with but it has been reworded.

**Comment by bangelic** [comment]
If everything looks good, can we get this merged before the release?

File: guides/common/modules/proc_upgrading-the-external-database.adoc [hunk 22]
```diff
@@ -0,0 +1,29 @@
...
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# {foreman-installer} \
+--foreman-db-host _newpostgres.example.com_ \
+--katello-candlepin-db-host _newpostgres.example.com_ \
+--foreman-proxy-content-pulpcore-postgresql-host _newpostgres.example.com_
+----
+. Copy the backup database to the new host.
+. Restore the backup database on the new host.
```
Comment (evgeni) [inline guides/common/modules/proc_upgrading-the-external-database.adoc:29]: These steps are already done above
```suggestion
```

File: guides/common/modules/proc_upgrading-the-external-database.adoc [hunk 23]
```diff
@@ -0,0 +1,29 @@
...
+You can upgrade an external database from {EL} 7 to {EL} 8 while upgrading {Project} from {ProjectVersionPrevious} to {ProjectVersion}.
+
+.Prerequisites
+* Create a new {EL} 8 based host for PostgreSQL server that follows the external database on {EL} 8 documentation.
+ifdef::satellite[]
+For more information, see https://access.redhat.com/documentation/en-us/red_hat_satellite/{ProjectVersion}/html/installing_satellite_server_in_a_connected_network_environment/performing-additional-configuration#using-external-databases_satellite[Using External Databases with {Project}].
+endif::[]
+ifdef::katello[]
+For more information, see https://docs.theforeman.org/{ProjectVersion}/Installing_Server/index-katello.html#using-external-databases_foreman[Using External Databases with {Project}].
+endif::[]
```
Comment (evgeni) [inline guides/common/modules/proc_upgrading-the-external-database.adoc:13]: This can be written using macros in one go:
```suggestion
ifdef::katello,orcharhino,satellite[]
For more information, see {InstallingServerDocURL}using-external-databases_{project-context}[Using External Databases with {Project}].
endif::[]
```

(we do not include external DB for foreman builds, no idea why

**Review by evgeni — CHANGES_REQUESTED** [review]

**Review by evgeni — APPROVED** [review]
Ack on the content
Should be squashed on merge

**Comment by evgeni** [comment]
I am away from my ssh key, so not merging. Whoever has the time feel free to!

**Comment by bangelic** [comment]
> I am away from my ssh key, so not merging. Whoever has the time feel free to!

@Lennonka or @adamlazik1 or @ekohl can anyone merge this today? Thanks in advance!

**Comment by maximiliankolb** [comment]
Merged to master and cherry-picked: ✅ 
   509be8f45..83c078c15  3.5 -> 3.5
   11e75a21b..cfacd0eaa  3.4 -> 3.4
   b087eb8fe..2c5b16274  3.3 -> 3.3
   8189ce835..a91479a4a  3.2 -> 3.2
   f33edb313..fed1766cc  3.1 -> 3.1

File: guides/common/modules/proc_upgrading-the-external-database.adoc (hunk 23 above)
Comment (ekohl) [reply guides/common/modules/proc_upgrading-the-external-database.adoc]: > (we do not include external DB for foreman builds, no idea why

8710bde2c7185c057d7f9a71313d7c166dae5fda. When I was reading through it, it only worked if you had Katello. If we add more conditionals to hide the Katello-specifics (like candlepin & pulp's databases) it can be included.
